docker/test/upgrade/run.sh

```diff
@@ -0,0 +1,514 @@
+        || echo -e "No OOM messages in clickhouse-server.log$OK" >> /test_output/test_results.tsv
+
+    # Logical errors
+    echo "Check for Logical errors in server log:"
+    rg -Fa -A20 "Code: 49, e.displayText() = DB::Exception:" /var/log/clickhouse-server/clickhouse-server.*.log > /test_output/logical_errors.txt \
+        && echo -e "Logical error thrown (see clickhouse-server.log or logical_errors.txt)$FAIL$(head_escaped /test_output/logical_errors.txt)" >> /test_output/test_results.tsv \
+        || echo -e "No logical errors$OK" >> /test_output/test_results.tsv
+
+    # Remove file logical_errors.txt if it's empty
+    [ -s /test_output/logical_errors.txt ] || rm /test_output/logical_errors.txt
+
+    # Crash
+    rg -Fa "########################################" /var/log/clickhouse-server/clickhouse-server.*.log > /dev/null \
+        && echo -e "Killed by signal (in clickhouse-server.log)$FAIL" >> /test_output/test_results.tsv \
+        || echo -e "Not crashed$OK" >> /test_output/test_results.tsv
+
+    # It also checks for crash without stacktrace (printed by watchdog)
+    echo "Check for Fatal message in server log:"
+    rg -Fa " <Fatal> " /var/log/clickhouse-server/clickhouse-server.*.log > /test_output/fatal_messages.txt \
+        && echo -e "Fatal message in clickhouse-server.log (see fatal_messages.txt)$FAIL$(trim_server_logs fatal_messages.txt)" >> /test_output/test_results.tsv \
+        || echo -e "No fatal messages in clickhouse-server.log$OK" >> /test_output/test_results.tsv
+
+    # Remove file fatal_messages.txt if it's empty
+    [ -s /test_output/fatal_messages.txt ] || rm /test_output/fatal_messages.txt
+
+    rg -Fa "########################################" /test_output/* > /dev/null \
+        && echo -e "Killed by signal (output files)$FAIL" >> /test_output/test_results.tsv
+
+    tar -chf /test_output/coordination.tar /var/lib/clickhouse/coordination ||:
+    for table in query_log trace_log
+    do
+        clickhouse-local --path /var/lib/clickhouse/ --only-system-tables -q "select * from system.$table format TSVWithNamesAndTypes" \
+          | zstd --threads=0 > /test_output/$table.tsv.zst ||:
+    done
+fi
+
+dmesg -T > /test_output/dmesg.log
+
+# OOM in dmesg -- those are real
+grep -q -F -e 'Out of memory: Killed process' -e 'oom_reaper: reaped process' -e 'oom-kill:constraint=CONSTRAINT_NONE' /test_output/dmesg.log \
+    && echo -e "OOM in dmesg$FAIL$(head_escaped /test_output/dmesg.log)" >> /test_output/test_results.tsv \
+    || echo -e "No OOM in dmesg$OK" >> /test_output/test_results.tsv
+
+mv /var/log/clickhouse-server/stderr.log /test_output/
+
+# If we failed to clone repo or download previous release packages,
+# we don't have any packages installed, but we need clickhouse-local
+# to be installed to create check_status.tsv.
+if ! command -v clickhouse-local &> /dev/null
+then
+    install_packages package_folder
```

Comment:
In this case, we could simply  `echo -e 'failure\tFailed to download packages' > /test_output/check_status.tsv` and exit immediately after that failure, but okay

Reply:
Agree, it will be better. Done